src/constants.ts

```diff
@@ -1,39 +1,65 @@
-)
-export const VERSION = packageXml.version
-
-export const TEST_CASE_CONFIG = join(process.cwd(), 'test-case.config.yaml')
+// =================================================================================================
+// Define the names of the environment variables that will be retireved.
+export const TEST_CASE_CONFIG_PATH_ENVAR_NAME = 'YATM_TEST_CASE_CONFIG_PATH'
+export const REQUIREMENTS_DIRECTORY_PATH_ENVAR_NAME =
+  'YATM_REQUIREMENTS_DIRECTORY_PATH'
+export const OUTPUT_DIRECTORY_PATH_ENVAR_NAME = 'YATM_OUTPUT_DIRECTORY_PATH'
 
-const REQUIREMENTS_DIRECTORY = 'requirements'
-const OUTPUT_DIRECTORY = 'generated-files'
-const TEST_CASES_DIRECTORY = 'test-cases'
-const RENDERED_TEST_CASE_DIRECTORY = 'rendered-test-cases'
+export const GITHUB_REPO_OWNER_ENVAR_NAME = 'GITHUB_REPO_OWNER'
+export const GITHUB_REPO_NAME_ENVAR_NAME = 'GITHUB_REPO_NAME'
+export const GITHUB_TOKEN_ENVAR_NAME = 'GITHUB_TOKEN'
+export const CODE_URL = 'https://github.com/audrow/yatm'
 
-export const INPUT_REQUIREMENTS_PATH = join(
-  process.cwd(),
-  REQUIREMENTS_DIRECTORY,
+// =================================================================================================
+// Retrieve the environment variables and define other constants to export.
+export const TEST_CASE_CONFIG = extractStringEnvVar(
+  TEST_CASE_CONFIG_PATH_ENVAR_NAME,
 )
-export const OUTPUT_PATH = join(process.cwd(), OUTPUT_DIRECTORY)
-export const OUTPUT_REQUIREMENTS_PATH = join(
-  OUTPUT_PATH,
-  REQUIREMENTS_DIRECTORY,
+export const INPUT_REQUIREMENTS_PATH = extractStringEnvVar(
+  REQUIREMENTS_DIRECTORY_PATH_ENVAR_NAME,
 )
-export const OUTPUT_TEST_CASE_PATH = join(OUTPUT_PATH, TEST_CASES_DIRECTORY)
+
+export const OUTPUT_PATH = extractStringEnvVar(
+  OUTPUT_DIRECTORY_PATH_ENVAR_NAME,
+  join(process.cwd(), 'generated-files'),
+)
+export const OUTPUT_REQUIREMENTS_PATH = join(OUTPUT_PATH, 'requirements')
+export const OUTPUT_TEST_CASE_PATH = join(OUTPUT_PATH, 'test-cases')
 export const OUTPUT_TEST_CASE_RENDER_PATH = join(
   OUTPUT_PATH,
-  RENDERED_TEST_CASE_DIRECTORY,
+  'rendered-test-cases',
 )
 
-export const TRANSLATION_MAP = {
-  jammy: 'Ubuntu Jammy',
-  installType: 'Install type',
-  fastdds: 'FastDDS',
-  dds: 'DDS vendor',
+export const GITHUB_TOKEN = extractStringEnvVar(GITHUB_TOKEN_ENVAR_NAME)
+export const GITHUB_RETRY_SECONDS = 60 * 5
+export const REPOSITORY: Repo = {
+  owner: extractStringEnvVar(GITHUB_REPO_OWNER_ENVAR_NAME),
+  name: extractStringEnvVar(GITHUB_REPO_NAME_ENVAR_NAME),
 }
 
-export const CODE_URL = 'https://github.com/audrow/yatm'
+const packageXml = JSON.parse(
+  fs.readFileSync(join(process.cwd(), 'package.json'), 'utf8'),
+)
+export const VERSION = packageXml.version
+
+// =================================================================================================
+// Helper function to retrieve environment variables or throw and error if unspecified.
+export function extractStringEnvVar(
```

Comment:
There's a really great library called "zod" that helps with this type of thing. Here's a post on how to use it to parse environment variables.

https://sergiodxa.com/articles/using-zod-to-safely-read-env-variables

Reply:
Thanks for the suggestion https://github.com/audrow/yatm/pull/404/commits/45c6ed312d26e7fc966c4d8943fbd6ecf761c641